Eastern-Research-Group/mywaterway
# Feature/update packages

## Related Issues:
* https://app.breeze.pm/projects/100762/cards/3820578

## Main Changes:
* Updated all dependencies in the project to their latest versions. Not all security issues are resolved by this. Some will remain until the relevant packages release their own updates. 
* Resolved new warnings in client code created by updating react versions.

## Steps To Test:
1. Run `npm install` in the app, client, and server directories.
2. Verify all Cypress tests are passing.
3. Verify there are no errors or warnings in the terminal or console.

## TODO:
* [ ] I did not update Cypress from 6.5 => 7.2 because it seemed like there are some breaking changes with version 7 and there is a deadline for this release.
